### PR TITLE
CORE-2395 Remove serialisation of private properties

### PR DESCRIPTION
**Pull request by kyriathar**
Prevents AMQP serialization of private properties.

- ignores property, during constructor properties information creation, if the property gets initialized through constructor and its getter is missing.
- removes redundant APIs from this change.
- aligns tests.

### CODE REVIEW

File: libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/model/LocalPropertyInformation.kt [hunk 1]
```diff
@@ -20,43 +23,29 @@ sealed class LocalPropertyInformation(val isCalculated: Boolean) {
 
     /**
      * A property of an interface, for which we have only a getter method.
-     *
-     * @param observedGetter The method which can be used to obtain the value of this property from an instance of its owning type.
```
Comment (josephzunigadaly) [inline libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/model/LocalPropertyInformation.kt:24]: Why did we delete this?

File: libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/model/LocalPropertyInformation.kt [hunk 2]
```diff
@@ -20,43 +23,29 @@ sealed class LocalPropertyInformation(val isCalculated: Boolean) {
...
      */
-    data class ReadOnlyProperty(val observedGetter: Method, override val type: LocalTypeInformation, override val isMandatory: Boolean) : LocalPropertyInformation(false)
+    data class ReadOnlyProperty(override val observedGetter: Method, override val type: LocalTypeInformation, override val isMandatory: Boolean) : LocalPropertyInformation(false)
 
     /**
      * A property for which we have both a getter, and a matching slot in an array of constructor parameters.
      *
-     * @param observedGetter The method which can be used to obtain the value of this property from an instance of its owning type.
```
Comment (josephzunigadaly) [inline libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/model/LocalPropertyInformation.kt:31]: I'm not sure why we are deleting this `observedGetter` part of the doc comment and leaving the rest of the comment intact. It appears to be correct even after the change.

If we restore this `@param observedGetter` line, does it fix the diff here in github?

File: libs/serialization/serialization-amqp/src/test/java/net/corda/internal/serialization/amqp/DummyOptional.java [hunk 3]
```diff
@@ -13,7 +13,7 @@ public boolean isPresent() {
         return item != null;
     }
 
-    public T get() {
+    public T getItem() {
```
Comment (josephzunigadaly) [inline libs/serialization/serialization-amqp/src/test/java/net/corda/internal/serialization/amqp/DummyOptional.java:16]: Is this change needed?

Comment (josephzunigadaly) [reply libs/serialization/serialization-amqp/src/test/java/net/corda/internal/serialization/amqp/DummyOptional.java:16]: I see, it needs to match the parameter name in the constructor

**Review by josephzunigadaly — DISMISSED** [review]

File: libs/serialization/serialization-amqp/src/test/java/net/corda/internal/serialization/amqp/DummyOptional.java (hunk 3 above)
Comment (kyriathar) [reply libs/serialization/serialization-amqp/src/test/java/net/corda/internal/serialization/amqp/DummyOptional.java:16]: Yes this change provides the getter for property `item` which is required so it becomes AMQP serializable.

File: libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/model/LocalPropertyInformation.kt (hunk 2 above)
Comment (kyriathar) [reply libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/model/LocalPropertyInformation.kt:31]: `observedGetter` was promoted as `LocalPropertyInformation` abstract property because it is now being used by all sub types.

File: libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/model/LocalPropertyInformation.kt (hunk 1 above)
Comment (kyriathar) [reply libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/model/LocalPropertyInformation.kt:24]: I think [answer](https://github.com/corda/corda-runtime-os/pull/2075#discussion_r984509640) applies here as well.

**Review by rick-r3 — DISMISSED** [review]
Looks right to me

**Review by josephzunigadaly — APPROVED** [review]
